### PR TITLE
leaktest: don't swallow stacks for panics

### PR DESCRIPTION
**Pull request by tbg**
leaktest recovers panics, but in translating them into a `t.Fatal` it
failed to report the stack trace.

Release note: None

### CODE REVIEW

File: pkg/util/leaktest/leaktest.go [hunk 1]
```diff
@@ -99,7 +100,12 @@ func AfterTest(t testing.TB) func() {
 		// If there was a panic, "leaked" goroutines are expected.
 		if r := recover(); r != nil {
 			atomic.StoreUint32(&leakDetectorDisabled, 1)
-			t.Fatalf("panic: %+v", r)
+			// NB: we don't want to re-panic here, for the Go test
+			// harness will not recover that for us. Instead, it will
+			// stop running the tests, but we are deciding here to fail
+			// only that one test but keep going otherwise.
+			// We need to explicitly print the stack trace though.
+			t.Fatalf("%v\n%s", r, debug.Stack())
```
Comment (irfansharif) [inline pkg/util/leaktest/leaktest.go:108]: Does it show the "panic" prefix by default? Think it's worth adding, it makes it grepable.

**Review by irfansharif — APPROVED** [review]

**Review by knz — APPROVED** [review]
Nice find!

File: pkg/util/leaktest/leaktest.go (hunk 1 above)
Comment (tbg) [reply pkg/util/leaktest/leaktest.go:108]: It does, that's why I removed it here.

**Comment by tbg** [comment]
@knz I introduced this in the first place :see_no_evil: 

Will hold off on the merge until the CI pipeline works again.

**Comment by tbg** [comment]
bors r=knz,irfansharif

**Pull request merged**